config/software/python.rb

```diff
@@ -37,6 +37,7 @@
   }
 
   python_configure = ["./configure",
+                      "--enable-unicode=ucs4",
```

Comment:
can we do this on Linux only? The default python on macOS is still built with `ucs2`, and the homebrew-provided python too, see https://github.com/Homebrew/homebrew-core/issues/12510